cmd/oceanbench/api.go

```diff
@@ -91,6 +91,40 @@ func apiHandler(w http.ResponseWriter, r *http.Request) {
 			fmt.Fprint(w, string(enc))
 			return
 
+		case "devices":
+			switch val {
+			case "site":
+				// Check that the user has at least read access.
+				if len(strings.Split(p.Data, ":")) != 2 {
+					writeHttpError(w, http.StatusBadRequest, "no site data in profile")
+					return
+				}
+				skey, err := strconv.ParseInt(strings.Split(p.Data, ":")[0], 10, 64)
+				if err != nil {
+					writeHttpError(w, http.StatusBadRequest, "invalid site data in profile data: %s", p.Data)
+					return
+				}
+				user, err := model.GetUser(ctx, settingsStore, skey, p.Email)
+				if err != nil {
+					writeHttpError(w, http.StatusInternalServerError, "unable to get user: %v", err)
+					return
+				}
+				if user.Perm&model.ReadPermission != 0 {
```

Comment:
Just check for user.Perm&model.ReadPermission == 0 and leave early so we don't have to nest.